### PR TITLE
Update Topics in UDP bridge

### PR DESCRIPTION
**Pull request by JanNiklasFeld**
Update UDP bridge topics of messages.
UDP bridge was already migrated and is working.
We checked, if all the written topics are still used and deleted unused ones.

See: https://github.com/bit-bots/udp_bridge/issues/20

## Necessary checks
- [ ] Update package version
- [ ] Run `colcon build`
- [ ] Write documentation
- [ ] Create issues for future work
- [ ] Test on your machine
- [ ] Test on the robot

### CODE REVIEW

**Review by timonegk — DISMISSED** [review]

File: config/udp_bridge.yaml [hunk 1]
```diff
@@ -14,20 +14,14 @@
       - /debug/dsd/body_behavior
       - /debug/dsd/head_behavior
       - /debug/dsd/hcm
-      - /move_base_simple/goal
```
Comment (Flova) [inline config/udp_bridge.yaml:17]: Replace this with our path planning goal

Comment (Flova) [reply config/udp_bridge.yaml:17]: You also might want to add the pose with covariance of the localization.

**Review by Flova — CHANGES_REQUESTED** [review]

**Comment by texhnolyze** [comment]
I'm just gonna hijack this branch/PR.

**Review by Flova — DISMISSED** [review]

**Comment by texhnolyze** [comment]
I've decided to extract the target ip for the monitoring host into the `game_settings.yaml` and get it via the `parameter_blackboard` (see https://github.com/bit-bots/bitbots_misc/pull/225).
As this lead to issues with execution/callbacks with the usage of `get_parameters_from_other_node` the sender was then additionally adapted to explicitly use and executor/timer instead of sleeps.

File: config/udp_bridge.yaml [hunk 2]
```diff
@@ -1,33 +1,27 @@
...
       - /gamestate
-      - /animation/goal
-      - /animation/cancel
-      - /animation/status
-      - /animation/result
-      - /animation/feedback
       - /animation
       - /diagnostics_agg
-      - /joint_states
+      - /joint_states
```
Comment (jaagut) [inline config/udp_bridge.yaml]:
```suggestion
      - /joint_states
```

File: udp_bridge/receiver.py [hunk 3]
```diff
@@ -1,45 +1,43 @@
...
-        self.node.get_logger().info("Initializing udp_bridge on port " + str(port))
+        port: str = node.get_parameter("port").value
+        self.node.get_logger().info(f"Initializing udp_bridge on port {port}")
 
         self.sock = socket.socket(type=socket.SOCK_DGRAM)
         self.sock.bind(("0.0.0.0", port))
         self.sock.settimeout(1)
 
-        self.known_senders = []  # type: list
+        self.known_senders: list[str] = []  # type: list
```
Comment (jaagut) [inline udp_bridge/receiver.py]:
```suggestion
        self.known_senders: list[str] = []
```

File: udp_bridge/receiver.py [hunk 4]
```diff
@@ -99,12 +95,18 @@ def validate_params(node: Node) -> bool:
     return result
 
 
+def run_spin_in_thread(node):
+    # Necessary in ROS2, else we get stuck
```
Comment (jaagut) [inline udp_bridge/receiver.py]:
```suggestion
    # Necessary in ROS 2, or else we get stuck
```

File: launch/send_standalone.launch [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<launch>
+    <!-- load the robot description -->
+    <include file="$(find-pkg-share bitbots_utils)/launch/base.launch" />
+
+    <node pkg="udp_bridge" exec="sender" output="screen">
+        <param from="$(find-pkg-share udp_bridge)/config/udp_bridge.yaml"/>
+    </node>
```
Comment (jaagut) [inline launch/send_standalone.launch]: Instead of copying the node, please include the `send.launch`.

File: udp_bridge/sender.py [hunk 6]
```diff
@@ -116,61 +102,81 @@ def validate_params(node:Node):
...
-        targets = node.get_parameter('target_ips').value
-        max_queue_size = node.get_parameter('sender_queue_max_size').value
-        topics = node.get_parameter("topics").value
+class UdpBridgeSender:
+    def __init__(self, node: Node):
+        self.node = node
+        self.freq: float = node.get_parameter("send_frequency").value
+
+        params_blackboard = get_parameters_from_other_node(node, "parameter_blackboard", ["monitoring_host_ip"])
+        self.target: str = params_blackboard["monitoring_host_ip"]
```
Comment (jaagut) [inline udp_bridge/sender.py]:
```suggestion
        target_ip_parameter_name: str = "monitoring_host_ip"
        params_blackboard = get_parameters_from_other_node(node, "parameter_blackboard", [target_ip_parameter_name])
        self.target: str = params_blackboard[target_ip_parameter_name]
```

File: udp_bridge/sender.py [hunk 7]
```diff
@@ -116,61 +102,81 @@ def validate_params(node:Node):
...
+        self.sock = self.setup_udp_broadcast_socket()
 
+        topics: list[str] = node.get_parameter("topics").value
+        max_queue_size: int = node.get_parameter("sender_queue_max_size").value
+        message_handler = self.setup_message_handler()
+        self.subscribers: list[AutoSubscriber] = list(
+            map(lambda topic: AutoSubscriber(topic, max_queue_size, message_handler, node), topics)
+        )
+
+    def setup_udp_broadcast_socket(self) -> socket.socket:
```
Comment (jaagut) [inline udp_bridge/sender.py]: Why do we want to broadcast? Don't we wanna send only to the target ip? Or is this just necessary socket-lingo?

File: udp_bridge/sender.py [hunk 8]
```diff
@@ -116,61 +102,81 @@ def validate_params(node:Node):
...
+def main():
+    rclpy.init()
+    node = Node("udp_bridge_sender", automatically_declare_parameters_from_overrides=True)
+
+    if validate_params(node):
+        sender = UdpBridgeSender(node)
 
-                except Empty:
-                    pass
+        exec = SingleThreadedExecutor()
```
Comment (jaagut) [inline udp_bridge/sender.py:177]: Do we REALLY want to use the single threaded executor?!?

Comment (texhnolyze) [reply udp_bridge/sender.py:177]: It seemed to work fine with the `SingleThreadedExecutor`, which also has a lot less overhead compared  to the `MultiThreadedExecutor`.
I'll test some more with the whole stack, but unless it becomes an issue I would prefer to keep the `SingleThreadedExecutor`.

**Review by texhnolyze — APPROVED** [review]

**Pull request merged**